habitat-baselines/habitat_baselines/common/env_factory.py

```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+from habitat import VectorEnv
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+
+class VectorEnvFactory(ABC):
```

Comment:
VectorEnvFactory seem to only have one implementation. Is this overengineered?

Reply:
I agree it only has 1 implementation in the current setup. However, this makes it much easier to (1) integrate with external environments that have a custom vectorized environment implementation, (2) create custom vector env logic.